### PR TITLE
Add AAAA Support

### PR DESCRIPTION
**Pull request by jriggins**

### CODE REVIEW

**Review by parkr — COMMENTED** [review]
Hi! I know this is still a work in progress but wanted to drop a comment about the 192 addresses. :)

File: lib/github-pages-health-check/domain.rb [hunk 1]
```diff
@@ -70,13 +70,30 @@ class Domain < Checkable
         "192.30.252.154"
       ].freeze
 
+      # TODO: Should these be added?
+      # I see them at https://api.github.com/meta, but some of the
+      # IPs are explictly test against in specs
+      # 192.30.252.153
+      # 192.30.252.154
```
[the remark this answers is not in the document]
Comment (jriggins) [reply lib/github-pages-health-check/domain.rb:77]: Thanks for the insight!

File: lib/github-pages-health-check/domain.rb [hunk 2]
```diff
@@ -204,7 +214,7 @@ def dns_zone_ns?
         end
       end
 
-      # Should the domain use an A record?
+      # Should the domain use an A or AAAA record?
```
Comment (jcudit) [inline lib/github-pages-health-check/domain.rb:217]: Is it safe to leave the method name this way given the change in scope?

**Review by jcudit — COMMENTED** [review]
👍🏾 on most of this but will defer an approval after a second read as this is a good chunk to review.

Specifically, the tests look good and the updates to `update-cdn-ips` are 👌🏾 too.  I'm still weighing the pros and cons around merging A and AAAA checks into single functions as it makes for an easier implementation but increases risk around misuse of the API.

File: lib/github-pages-health-check/domain.rb (hunk 2 above)
Comment (jriggins) [reply lib/github-pages-health-check/domain.rb:217]: Yeah it's a great observation and I did ponder a little. I leaned on the notion that I _believe_ that A and AAAA records are serving the similar purposes, but I can look to separate those instead if I'm conflating them too much.

**Comment by jriggins** [comment]
ℹ️  Going to make some updates.

**Comment by jriggins** [comment]
Ready for another review. 🚀

File: script/update-cdn-ips [hunk 3]
```diff
@@ -8,15 +8,43 @@ require "open-uri"
...
-  puts "Fetching #{url}..."
-  data = open(url).read
-  data = JSON.parse(data)["addresses"].join("\n") if source == :fastly
   File.write(file, data)
+  puts "Writing contents to #{file} and staging changes."
   `git add --verbose #{file}`
 end
+
+def parse_cdn_response(source, ips)
+  send("parse_#{source}", ips)
```
Comment (TooManyBees) [inline script/update-cdn-ips:39]: 🍜 not a blocker, but I think it's more readable here to
```ruby
case source
when "fastly"
  # ...
when "cloudflare"
  # ...
else
  raise "oh no #{source}"
end
```

**Review by TooManyBees — DISMISSED** [review]
code looks 👍🏼

File: lib/github-pages-health-check/domain.rb [hunk 4]
```diff
@@ -517,7 +538,9 @@ def scheme
       def cdn_ip?(cdn)
         return unless dns?
 
-        a_records = dns.select { |answer| answer.type == Dnsruby::Types::A }
+        a_records = dns.select do |answer|
```
Comment (jcudit) [inline lib/github-pages-health-check/domain.rb:541]: nit: would rename to `addresses`

**Review by jcudit — APPROVED** [review]

File: lib/github-pages-health-check/domain.rb [hunk 5]
```diff
@@ -78,10 +78,10 @@ class Domain < Checkable
...
 
       CURRENT_IPV6_ADDRESSES = %w(
-        2606:50c0:8000::1538
-        2606:50c0:8001::1538
-        2606:50c0:8002::1538
-        2606:50c0:8003::1538
+        2606:50c0:8000::153
+        2606:50c0:8001::153
+        2606:50c0:8002::153
+        2606:50c0:8003::153
```
Comment (tcbyrd) [inline lib/github-pages-health-check/domain.rb:84]: 👍 , but just for confirmation:

```sh
curl https://api.github.com/meta | jq .pages | grep ::

  "2606:50c0:8000::153/128",
  "2606:50c0:8001::153/128",
  "2606:50c0:8002::153/128",
  "2606:50c0:8003::153/128"
```

**Review by tcbyrd — APPROVED** [review]

File: lib/github-pages-health-check/domain.rb (hunk 5 above)
Comment (jriggins) [reply lib/github-pages-health-check/domain.rb:84]: Thanks. Yeah I bet I had a bad block cut on the CIDR and left the `8` 🙈

**Pull request merged**